### PR TITLE
feat(plugin_toc): support emphasis such as `*Hello world*` `***Hello world***` and  `_Hello world_`

### PR DESCRIPTION
**Pull request by shulaoda**
Fix this issue [#952](https://github.com/web-infra-dev/rspress/issues/952) in `rspress`

### CODE REVIEW

File: crates/plugin_toc/src/lib.rs [hunk 1]
```diff
@@ -298,6 +338,7 @@ mod tests {
 
     assert_eq!(result.title, "Hello`World`");
     assert_eq!(result.toc.len(), 4);
+    print!("{:?}", result.toc);
```
Comment (Timeless0911) [inline crates/plugin_toc/src/lib.rs:341]: can be deleted?

Comment (shulaoda) [reply crates/plugin_toc/src/lib.rs:341]: I'm sorry for my mistake.

**Comment by shulaoda** [comment]
Should we return to the original format, like *\*Hello World\**, and plugin_toc will reutrn \*\*Hello World\*\* which is rendered like    **Hello World** in the theme-default toc.
Is it necessary to render italics and bold in the sidebar? Should we standardize user behavior or directly return labels in mdx-rs, just like `*/* Hello World/**` returning `<em>*Hello World*</em>`.

**Comment by shulaoda** [comment]
> Should we return to the original format, like _*Hello World*_, and plugin_toc will reutrn **Hello World** which is rendered like **Hello World** in the theme-default toc. Is it necessary to render italics and bold in the sidebar? Should we standardize user behavior or directly return labels in mdx-rs, just like `*/* Hello World/**` returning `\<em\>*Hello World*\</em\>`.

@Timeless0911

**Comment by SoonIter** [comment]
> Should we return to the original format, like *Hello World*, and plugin_toc will reutrn **Hello World** which is rendered like Hello World in the theme-default toc.

`**Hello World**` -> **Hello World**

I think this is a standard behavior, and we can follow it.

> Is it necessary to render italics and bold in the sidebar toc?

Yeah, we have already implemented it ... but `<em>Hello World</em>` seems we can not extract the inside text currently

image

The goal of this pr can be regarded as supporting `*Hello world*` `***Hello world***` and `_Hello world_`, I think this goal has been achieved in this mr. You can rebase it and I will approve it.

**Comment by shulaoda** [comment]
> > Should we return to the original format, like _Hello World_, and plugin_toc will reutrn **Hello World** which is rendered like Hello World in the theme-default toc.
> 
> `**Hello World**` -> **Hello World**
> 
> I think this is a standard behavior, and we can follow it.

What I mean is `*\*Hello World\**`, it should be rendered correctly as *\*Hello World\**.
But plugin_toc will reutrn`**Hello World**` which is rendered like **Hello World** in the theme-default toc.

**Comment by shulaoda** [comment]
> But we need to change the code togther with https://github.com/web-infra-dev/rspress/blob/695dea8b65f00e7533f26c6e37a9395fdcfcbbf2/packages/theme-default/src/logic/utils.ts#L47

Yes, I have submitted the relevant PR [#1012](https://github.com/web-infra-dev/rspress/pull/1012) in `web-infla-dev/rspress`, but it is currently closed. I need to wait for the PR confirmation here before aligning.

> And the remark-toc-plugin is also aligned as much as possible. Maybe we can split it into another pr.

Do you mean I need to open a new PR and replace the format in `plugin_toc`? For example: `* *` replaced with `<i> </i>`,  \` \` replaced with `<code> </code>`.
If so, will this conflict with other compilation results, such as `** **` being compiled as `<strong> </strong>`, does this mean that I need to make further replacements in the `rspress/theme-default(toc)`?

**Comment by SoonIter** [comment]
In the long run, we need to optimize the plugin-toc-related logic in both rust and js version in order to ensure that they have the same behavior, more test cases are needed. 

But let's pending this corner case `*\*Hello World\**` and merge the pr first

Then I will merge https://github.com/web-infra-dev/rspress/pull/1012 after we publish the latest version of mdx-rs.
